### PR TITLE
Inline svgs

### PR DESCRIPTION
**Pull request by sndrs**
This creates the mechanism by which we can move all svgs to inlining, with a single demo of the search icon in the header: http://www.browserstack.com/screenshots/81bb1914b39c17685205de38c706dae39ece6368.
# changes
- [x] svgs will now live in `src/inline-svgs/*.svg`
- [x] inlined svgs are wrapped in a span with class to address them by, prefixed with `.inline-` until old css versions are removed
- [x] inlined svgs should have no fill information
- [x] by default, inlined icon svgs are filled white
- [x] inlined svgs are svgo-optimised after copying, meaning svgs in `static` can contain useful meta data
- [ ] move svg helper classes to the svg element itself, not a wrapper?
- [ ] look into IE9 `fill` in css not working
# use

`@fragments.inlineSvg(filename [, type, classes])`
- you must supply a filename, which by default must be in the top level (`static/src/inline-svgs`)
- types are grouped into sub-directories (`static/src/inline-svgs/icon`)
- the filename and type are automatically added as classes on the svg:
  - `@fragments.inlineSvg("my-image")` -> `class="inline-my-image"`
  - `@fragments.inlineSvg("search-36", "icon")` -> `class="inline-search-36 inline-icon"`
- a 3rd argument takes a list of classnames:
  - `@fragments.inlineSvg("my-image", classes = List("class1", "class2"))` -> `class="inline-my-image inline-class1 inline-class2"`
# to note
- all inline svgs are optional - IE8/android 2.\* will not get fallbacks
- required images (e.g. logo) will be hand-tuned pngs (@1x and @2x) to minimise places where edits need to be made

### CODE REVIEW

File: common/app/assets/assets.scala [hunk 1]
```diff
@@ -5,7 +5,7 @@ import org.apache.commons.io.IOUtils
 import play.api.{ Mode, Play }
 import play.api.libs.json.{ JsString, Json, JsObject }
 import conf.Configuration
-import collection.mutable.{ Map => MutableMap }
```
Comment (robertberry) [inline common/app/assets/assets.scala]: I'm glad you've used `ConcurrentMap` here ... I'm guessing this other `MutableMap` should be a `ConcurrentMap`, too, though. Although that could be sorted in a separate refactor pull request.

Comment (robertberry) [inline common/app/assets/assets.scala]: Oh wait, you did delete this. My brain hurts :(

**Comment by robertberry** [comment]
As ever, good work, chaps :+1:

File: common/app/assets/assets.scala (hunk 1 above)
Comment (gklopper) [inline common/app/assets/assets.scala]: Yeah, I wondered if that caused some of our (very seldom) box does not start up problems leading to longer deploys.

File: common/app/conf/switches.scala [hunk 2]
```diff
@@ -355,7 +355,7 @@ object Switches {
 
   val SearchSwitch = Switch("Feature", "google-search",
     "If this switch is turned on then Google search is added to the sections nav.",
-    safeState = Off, sellByDate = never
+    safeState = On, sellByDate = never
```
Comment (phamann) [inline common/app/conf/switches.scala]: Is this meant to be in this PR?

**Comment by gklopper** [comment]
lets ship it

**Comment by phamann** [comment]
Out of interest, why are you using `svgmin` instead of `svgo`?

File: common/app/conf/switches.scala (hunk 2 above)
Comment (gklopper) [inline common/app/conf/switches.scala]: Yeah, I purposefully checked that in

**Comment by sndrs** [comment]
grunt-svgo deprecated in favour of grunt-svgmin, which uses svgo https://github.com/svg/svgo-grunt

**Comment by phamann** [comment]
Ah nice!

**Comment by paperboyo** [comment]
Wow!

This is cool! That means no more duplicated svgs for different colours, right? And possibly even animations (I really, really, haven't said that! ;).

I have one question. If the inlines are auto-optimised from source files, with current (small) icon sizes there is a slight potential for garbled paths from ``floatPrecision: 1`` (assuming that's a part of our config for svgo) and also svgo sometimes makes a mess of a graphic anyway.

The second thing would mean that we would have to observe the inline result every time a new svg is added (or switch off all the potentially offending plugins in svgo for safety which would make the files bigger than necessary). [The usual paperboyo's overthinking would suggest automatic glitch finding robot comparing a source and i't corresponding optimised inline every time a new one is added. That robot could then be used by mediaservice when optimising other svgs, like the maps produced by Graphics].

But the first thing (rounding errors) maybe could be avoided by keeping the source files bigger than now and abstracting the width and height (using them smaller on a page). E.g. Workflow uses 128x128px for all icons and ``floatPrecision: 1`` does them no harm (I've checked).

Apologies if all of that was already taken into account, but my plunges into the code almost never provide me with any answers...

Regards
Mateusz

p.s. gee, was the above really what you call 'one question', paperboyo?! :D

**Comment by sndrs** [comment]
@paperboyo yeah @sammorrisdesign and I were talking about doing exactly that (large svgs scaled down), it's a nice idea. won't go into this PR, but it's a good move.

in terms of the optimisation happening 'autonomously', it's just the devs job to check it optimises correctly before merging. i think there's a lot of upside in the source files being developer-friendly and making them machine/wire-friendly once they get to that domain.

**Comment by paperboyo** [comment]
:-)

Yeah, I agree that having more of an 'original' as source is always better.

Poor devs ;-) I am always eager to help with the files that will give headache, remember!

Thanks. On a totally unrelated note, I will try to talk to you about fonts, if I may, maybe tomorrow, if your time permits...

Regards
Mateusz

**Pull request merged**